components/Gallery/ArticleGallery.js

```diff
@@ -50,7 +48,7 @@ class ArticleGallery extends Component {
     this.state = {
       show: false,
       startItemSrc: null,
-      ...this.getDerivedStateFromProps(props)
+      ...ArticleGallery.getDerivedStateFromProps(props)
```

Comment:
Not needed anymore? because if it is static react should call it automatically before render (on initial and updates). Also remove from `componentWillReceiveProps` and check if a `shouldComponentUpdate` is necessary to prevent unnecessary recomputes. Or rewrite with `useMemo(() => { walkMdast() }, [props.article])`